src/auth/HTTPAuth.ts

```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+import { IAuthentication } from '../types/Authentication';
+
+interface IHTTPAuthOptions {
+	url: string;
+}
+
+export class HTTPAuth implements IAuthentication {
+	private url: string;
+
+	constructor(config: IHTTPAuthOptions) {
+		this.url = config.url;
+	}
+
+	async authenticate(username: string, password: string) {
+
+		let success = false;
+		try {
+			
+			await axios
+			  .post(this.url, {
+			    username: username,
+			    password: password
+			  })
```

Comment:
i would prefer "three" cases here:
1.) auth success
2.) auth failed
3.) other http error (e.g. server error like 5xx)

remove the try / catch block and just use
await axios..
but add the axios config:
```
validateStatus: function (status) {
  return status >= 200 && status < 500;
}
```

afterwards all 200 till 499 codes are "resolved" and will not throw, but 5xx will throw (as exepcted)


e.g. somethign like this (pseudo code):

```ts
const res = await axios
			  .post(this.url, {
			    username,
			    password
			  }, validateStatus: function (status) {
  return status >= 200 && status < 500;
});

if (res.status == 200) { 
  /// Return code 200 -> authentication successful.
  return true;
  }

return false;

```